includes/Handlers/Connection.php

```diff
@@ -141,17 +146,29 @@ public function handle_connect() {
 				throw new SV_WC_API_Exception( 'Invalid nonce' );
 			}
 
-			$access_token = ! empty( $_GET['access_token'] ) ? sanitize_text_field( $_GET['access_token'] ) : '';
+			$merchant_access_token = ! empty( $_GET['merchant_access_token'] ) ? sanitize_text_field( $_GET['merchant_access_token'] ) : '';
 
-			if ( ! $access_token ) {
+			if ( ! $merchant_access_token ) {
 				throw new SV_WC_API_Exception( 'Access token is missing' );
 			}
 
-			$access_token = $this->create_system_user_token( $access_token );
+			$system_user_access_token = ! empty( $_GET['system_user_access_token'] ) ? sanitize_text_field( $_GET['system_user_access_token'] ) : '';
+
+			if ( ! $system_user_access_token ) {
+				throw new SV_WC_API_Exception( 'System User access token is missing' );
+			}
 
-			$this->update_access_token( $access_token );
+			$system_user_id = ! empty( $_GET['system_user_id'] ) ? sanitize_text_field( $_GET['system_user_id'] ) : '';
 
-			$api = new \WC_Facebookcommerce_Graph_API( $access_token );
+			if ( ! $system_user_id ) {
+				throw new SV_WC_API_Exception( 'System User ID is missing' );
+			}
+
+			$this->update_access_token( $system_user_access_token );
+			$this->update_merchant_access_token( $merchant_access_token );
+			$this->update_system_user_id( $system_user_id );
+
+			$api = new \WC_Facebookcommerce_Graph_API( $merchant_access_token );
```

Comment:
I am probably confused, but I thought we would use the system user access token for everything. 🤔

Reply:
May be worth adding a comment above this line explaining why we use this token here, for visibility.

Reply:
You're right @deivamagalhaes , the goal is to use the system user access token, however, the API request to get assets IDs currently does not accept that token. I asked Connor and he was able to find a problem with the way permissions were being checked on that endpoint.

A fix is currently [being reviewed](https://a8c.slack.com/archives/C010ND8MRPU/p1592438962330500?thread_ts=1591237950.248100&cid=C010ND8MRPU), so we should be able to switch that line to use the system user access token soon.

I added a TODO note for that: 3cf2082